### PR TITLE
add tpu example for torchrun on GKE

### PR DESCRIPTION
**Pull request by chengcongdu**

### CODE REVIEW

**Comment by chengcongdu** [comment]
@will-cromar  PTAL

File: docs/kubernetes.md [hunk 1]
```diff
@@ -167,9 +167,150 @@ Max Accuracy: 100.00%
...
+          name: shm
+        env:
+        - name: PJRT_DEVICE
+          value: 'TPU'
+        - name: XLA_USE_BF16
+          value: '1'
+        - name: USE_TORCH
+          value: 'ON'
+        - name: JOB_NAME
+          value: 'torch-xla-tpu-2x4'
```
Comment (will-cromar) [inline docs/kubernetes.md:283]: Can you refer to the `job` label here too?

File: docs/kubernetes.md [hunk 2]
```diff
@@ -167,9 +167,150 @@ Max Accuracy: 100.00%
...
+        - containerPort: 8477
+        - containerPort: 8476
+        - containerPort: 8431 # Port to export TPU usage metrics, if supported.
+        volumeMounts:
+        - mountPath: /dev/shm
+          name: shm
+        env:
+        - name: PJRT_DEVICE
+          value: 'TPU'
+        - name: XLA_USE_BF16
```
Comment (will-cromar) [inline docs/kubernetes.md:278]: Please remove the `XLA_USE_BF16`. We're removing that option in the next release.

File: docs/kubernetes.md [hunk 3]
```diff
@@ -167,9 +167,150 @@ Max Accuracy: 100.00%
...
+        emptyDir:
+          medium: Memory
+      containers:
+      - name: training
+        image: us-central1-docker.pkg.dev/tpu-pytorch-releases/docker/xla:nightly_3.10_tpuvm
+        command:
+        - bash
+        - -cxue
+        - |
+          apt-get update -y
```
Comment (will-cromar) [inline docs/kubernetes.md:235]: The additional pip installs and symlinking should no longer be necessary with any recent release

File: docs/kubernetes.md [hunk 4]
```diff
@@ -167,9 +167,150 @@ Max Accuracy: 100.00%
...
+        - containerPort: 8431 # Port to export TPU usage metrics, if supported.
+        volumeMounts:
+        - mountPath: /dev/shm
+          name: shm
+        env:
+        - name: PJRT_DEVICE
+          value: 'TPU'
+        - name: XLA_USE_BF16
+          value: '1'
+        - name: USE_TORCH
```
Comment (will-cromar) [inline docs/kubernetes.md:280]: What is `USE_TORCH`?

File: docs/kubernetes.md [hunk 5]
```diff
@@ -167,9 +167,150 @@ Max Accuracy: 100.00%
...
+spec:
+  clusterIP: None
+  selector:
+    job-name: torch-xla-tpu-2x4
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: torch-xla-tpu-2x4
+  labels:
```
Comment (will-cromar) [inline docs/kubernetes.md:194]: nit: remove the empty `labels`

File: docs/kubernetes.md [hunk 6]
```diff
@@ -167,9 +167,150 @@ Max Accuracy: 100.00%
...
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: headless-svc
+  namespace: default
+spec:
+  clusterIP: None
+  selector:
+    job-name: torch-xla-tpu-2x4
```
Comment (will-cromar) [inline docs/kubernetes.md:188]: Can you change the selector to match on `headless-svc: "true"` like the above example?

File: docs/kubernetes.md [hunk 7]
```diff
@@ -167,9 +167,150 @@ Max Accuracy: 100.00%
...
+      restartPolicy: Never # look in https://kubernetes.io/docs/concepts/workloads/controllers/job/
+      subdomain: headless-svc
+      volumes:
+        # Increase size of tmpfs /dev/shm to avoid OOM.
+      - name: shm
+        emptyDir:
+          medium: Memory
+      containers:
+      - name: training
+        image: us-central1-docker.pkg.dev/tpu-pytorch-releases/docker/xla:nightly_3.10_tpuvm
```
Comment (will-cromar) [inline docs/kubernetes.md:230]: Please use a stable image for this example: `r2.3.0_3.10_tpuvm`

**Review by will-cromar — COMMENTED** [review]
Thanks!

File: docs/kubernetes.md (hunk 7 above)
Comment (chengcongdu) [reply docs/kubernetes.md:230]: without using the nightly image, it is causing libtpu error
F0000 00:00:1719966307.830579     588 pjrt_registry.cc:117] Non-OK-status: pjrt::LoadPjrtPlugin("tpu", tpu_library_path).status() status: INTERNAL: Failed to open libtpu.so: libtpu.so: cannot open shared object file: No such file or directory

And installing torch_xla[tpu] failed with version conflicts
ERROR: Cannot install torch-xla[tpu]==2.1.0, torch-xla[tpu]==2.2.0 and torch-xla[tpu]==2.3.0 because these package versions have conflicting dependencies.

File: docs/kubernetes.md (hunk 6 above)
Comment (chengcongdu) [reply docs/kubernetes.md:188]: done

File: docs/kubernetes.md (hunk 5 above)
Comment (chengcongdu) [reply docs/kubernetes.md:194]: done

File: docs/kubernetes.md (hunk 4 above)
Comment (chengcongdu) [reply docs/kubernetes.md:280]: removed

File: docs/kubernetes.md (hunk 2 above)
Comment (chengcongdu) [reply docs/kubernetes.md:278]: done

File: docs/kubernetes.md (hunk 3 above)
Comment (chengcongdu) [reply docs/kubernetes.md]: removed

File: docs/kubernetes.md (hunk 7 above)
Comment (will-cromar) [reply docs/kubernetes.md:232]: Sorry, I sent you the wrong link. Can you try this one that has libtpu preinstalled? `r2.3.0_libtpu_3.10_tpuvm`

Comment (chengcongdu) [reply docs/kubernetes.md]: updated

**Review by will-cromar — APPROVED** [review]
Thanks!
